File: cli-tools/automation/node/create-trimap.js
const axios = require('axios');
const { BASE_URL, getAccessToken } = require('./utils');

async function createTrimap() {
    const imageUrl = process.argv[2];

    if (!imageUrl) {
        console.log('Usage: node create-trimap.js <imageUrl>');
        process.exit(1);
    }

    const token = await getAccessToken();
    try {
        const response = await axios.post(`${BASE_URL}/images/create-trimap`, {
            image_url: imageUrl
        }, {
            headers: { 'Authorization': `Bearer ${token}` }
        });
        console.log(JSON.stringify(response.data, null, 2));
    } catch (error) {
        console.error('Error:', error.response ? error.response.data : error.message);
    }
}

createTrimap();

File: cli-tools/automation/python/create_trimap.py
import sys
import json
import requests
from utils import BASE_URL, get_access_token

def create_trimap():
    if len(sys.argv) < 2:
        print('Usage: python create_trimap.py <imageUrl>')
        sys.exit(1)

    image_url = sys.argv[1]

    token = get_access_token()
    headers = {'Authorization': f'Bearer {token}', 'Content-Type': 'application/json'}
    payload = {'image_url': image_url}
    
    try:
        response = requests.post(f"{BASE_URL}/images/create-trimap", json=payload, headers=headers)
        response.raise_for_status()
        print(json.dumps(response.json(), indent=2))
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    create_trimap()
